src/test/java/stashpullrequestbuilder/stashpullrequestbuilder/StashRepositoryTest.java

```diff
@@ -84,4 +95,156 @@ public void getTargetPullRequestsSkipsNullStatePullRequests() {
+    comment.setText("NO TEST");
+    List<StashPullRequestComment> comments = Collections.singletonList(comment);
+
+    when(stashApiClient.getPullRequests()).thenReturn(Collections.singletonList(pullRequest));
+    when(pullRequest.getState()).thenReturn("OPEN");
+    when(trigger.getCiSkipPhrases()).thenReturn("NO TEST");
+    when(pullRequest.getTitle()).thenReturn("Add some bloat");
+    when(pullRequest.getFromRef()).thenReturn(repository);
+    when(pullRequest.getToRef()).thenReturn(repository);
+    when(repository.getRepository()).thenReturn(repoRepo);
+    when(stashApiClient.getPullRequestComments(any(), any(), any())).thenReturn(comments);
+    when(project.getDisplayName()).thenReturn("Pull Request Builder Project");
+
+    assertThat(stashRepository.getTargetPullRequests(), empty());
+  }
+
+  @Test
+  public void getTargetPullRequestsSkipPhraseIsCaseInsensitive() {
+    when(stashApiClient.getPullRequests()).thenReturn(Collections.singletonList(pullRequest));
+    when(pullRequest.getState()).thenReturn("OPEN");
+    when(pullRequest.getTitle()).thenReturn("Disable any testing");
+    when(trigger.getCiSkipPhrases()).thenReturn("disable ANY Testing");
+
+    assertThat(stashRepository.getTargetPullRequests(), empty());
+  }
+
+  @Test
+  public void getTargetPullRequestsSkipPhraseMatchedAsSubstring() {
+    when(stashApiClient.getPullRequests()).thenReturn(Collections.singletonList(pullRequest));
+    when(pullRequest.getState()).thenReturn("OPEN");
+    when(pullRequest.getTitle()).thenReturn("This will get no testing whatsoever");
+    when(trigger.getCiSkipPhrases()).thenReturn("NO TEST");
+
+    assertThat(stashRepository.getTargetPullRequests(), empty());
+  }
+
+  @Test
+  public void getTargetPullRequestsSupportsMultipleSkipPhrasesAndPadding() {
+    when(stashApiClient.getPullRequests()).thenReturn(Collections.singletonList(pullRequest));
+    when(pullRequest.getState()).thenReturn("OPEN");
+    when(pullRequest.getTitle()).thenReturn("This will get no testing whatsoever");
+    when(trigger.getCiSkipPhrases()).thenReturn("\tuntestable , \n NO TEST\t, \r\ndon't worry!");
+
+    assertThat(stashRepository.getTargetPullRequests(), empty());
+  }
+
+  @Test
+  public void getTargetPullRequestsBuildsIfSkipPhraseIsEmpty() {
+    when(stashApiClient.getPullRequests()).thenReturn(Collections.singletonList(pullRequest));
+    when(pullRequest.getState()).thenReturn("OPEN");
+    when(pullRequest.getTitle()).thenReturn("NO TEST");
+    when(trigger.getCiSkipPhrases()).thenReturn("");
+    when(pullRequest.getFromRef()).thenReturn(repository);
+    when(pullRequest.getToRef()).thenReturn(repository);
+    when(repository.getRepository()).thenReturn(repoRepo);
+
+    assertThat(stashRepository.getTargetPullRequests(), contains(pullRequest));
+  }
+
+  @Test
+  public void getTargetPullRequestsBuildsIfSkipPhraseIsNull() {
+    when(stashApiClient.getPullRequests()).thenReturn(Collections.singletonList(pullRequest));
```

Comment:
there is a lot of repetition here, it would be nice to make it more DRY later

Reply:
I can do it, but it would affect the existing tests, so let do it separately.

I'm considering constructing simple data objects like I do with the branch. It's should be more readable and more amenable to refactoring that mocking every trivial dependency. I could create objects in `before()` and use them in tests, changing them if necessary.

Reply:
Done in #94.